cli/src/clients/cloud/client.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2024 -  Restate Software, Inc., Restate GmbH.
+// All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! A wrapper client for Restate Cloud HTTP service.
+
+use http::StatusCode;
+use serde::{de::DeserializeOwned, Serialize};
+use std::time::Duration;
+use thiserror::Error;
+use tracing::{debug, info};
+use url::Url;
+
+use crate::build_info;
+use crate::cli_env::CliEnv;
+
+use super::super::errors::ApiError;
+
+#[derive(Error, Debug)]
+#[error(transparent)]
+pub enum Error {
+    // Error is boxed because ApiError can get quite large if the message body is large.
+    Api(#[from] Box<ApiError>),
+    #[error("(Protocol error) {0}")]
+    Serialization(#[from] serde_json::Error),
+    Network(#[from] reqwest::Error),
+}
+
+/// A lazy wrapper around a reqwest response that deserializes the body on
+/// demand and decodes our custom error body on non-2xx responses.
+pub struct Envelope<T> {
+    inner: reqwest::Response,
+
+    _phantom: std::marker::PhantomData<T>,
+}
+
```

Comment:
Is it intentional to copy the code from metas_client or is it possible to reuse existing implementation?